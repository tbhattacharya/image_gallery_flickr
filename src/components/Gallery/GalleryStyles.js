export const styles = {
  root: {
    flexGrow: 1,
  },
  mainConatiner: {
    padding: '25px 80px'
  },
  subtitle: {
    textAlign: 'center',
    marginBottom: '50px',
    color: 'wheat'
  },
  overlayItem: {
    display: 'none',
    color: 'wheat',
    textAlign: 'center',
    'flex-direction': 'column',
    'justify-content': 'center',
    position: 'absolute',
    width: '100%',
    height: '100%',
    background: 'black',
    opacity: '0.7',
    'z-index': '2',
    transition: '.5s ease',
    top: '0',
    left: '0'
  },
  thumbnail: {
    position: 'relative',
    '&:hover $overlayItem': {
      display: 'flex'
    }
  },
  photo:{
    padding: '5px',
    backgroundColor: 'antiquewhite'
  }
};