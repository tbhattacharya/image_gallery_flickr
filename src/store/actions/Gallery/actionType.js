export const FETCH_GRID_SUCCESS = 'FETCH_GRID_SUCCESS';
export const FETCH_GRID_REQ = 'FETCH_GRID_REQ';
export const UPDATE_GALLERY_INFO = 'UPDATE_GALLERY_INFO';