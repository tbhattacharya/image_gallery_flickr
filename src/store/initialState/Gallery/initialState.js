export const initialState = {
    photoGrid: [],
    galleryInfo: {}
}