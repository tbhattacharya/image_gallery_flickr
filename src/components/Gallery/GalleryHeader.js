import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { styles } from './GalleryStyles';
import React from 'react';

class GalleryHeader extends React.Component {

    render() {
        const { classes, gallery } = this.props;
        return (
            <div className={classes.root}>
                <AppBar position="static" color="primary">
                    <Toolbar>
                        <Typography variant="title" color="inherit">
                            Gallery - {gallery && gallery.title ? gallery.title._content : ''}
                        </Typography>
                        <div>
                        <Typography variant="subheading" color="inherit">
                            &nbsp;&nbsp;Images by - {gallery ? gallery.owner_name : ''}
                        </Typography>
                        </div>
                    </Toolbar>
                </AppBar>
            </div>
        );
    }
}

export default withStyles(styles)(GalleryHeader);