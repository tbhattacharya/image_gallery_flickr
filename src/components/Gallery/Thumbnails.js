import React from 'react';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import { Thumbs } from '../../utils/Constants';
import 'react-lazy-load-image-component/src/effects/blur.css';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core/styles';
import { styles } from './GalleryStyles';

class Thumbnails extends React.Component {

    handleClick = ()=>{
        this.props.handleClick(this.props.image);
    }
    render() {
        const { image } = this.props;
        //Create image url for Lazy loader and return grid item element
        const src = `https://farm${image.farm}.staticflickr.com/${image.server}/${image.id}_${image.secret }.jpg`;
        return (
            <Grid item xs={3} classes={{ container: 'true' }} className={this.props.classes.thumbnail}
                onClick={this.handleClick}>
                <div className={this.props.classes.overlayItem} >
                    {image.title}
                </div>
                <div className={this.props.classes.photo}>
                    <LazyLoadImage
                        height={Thumbs.height}
                        src={src}
                        effect="blur"
                        width={Thumbs.width} />
                </div>
            </Grid>
        );
    }
}
export default withStyles(styles)(Thumbnails);