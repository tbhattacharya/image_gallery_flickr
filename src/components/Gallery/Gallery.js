import React from 'react';
import { isEmpty } from 'lodash';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect'
import * as actionType from '../../store/actions/Gallery/actionType';
import GalleryHeader from './GalleryHeader';
import { withStyles } from '@material-ui/core/styles';
import { styles } from './GalleryStyles';
import { forEach } from 'lodash';
import Thumbnails from './Thumbnails';
import Grid from '@material-ui/core/Grid';

class Gallery extends React.Component {

    constructor(props) {
        super(props);
    }

    componentDidMount() {
        if (isEmpty(this.props.photoGrid)) {
            this.props.fetchGalleryInformation('72157677476597845');
        }
    }

    getGalleryThumbnails = () => {
        let thumbs = [];
        forEach(this.props.photoGrid, photo => {
            thumbs.push(<Thumbnails image={photo} />);
        })
        return thumbs;
    }

    render() {
        const { galleryInfo, classes } = this.props;
        const description = (galleryInfo && galleryInfo.description) ? galleryInfo.description._content.replace(/\n/g, '<br />') : '';
        return (
            <div>
                <GalleryHeader gallery={galleryInfo} />
                <div className={classes.mainConatiner}>
                    <div className={classes.subtitle} contentEditable='true' dangerouslySetInnerHTML={{ __html: description }} ></div>
                    <Grid container spacing={24}>
                        {this.getGalleryThumbnails()}
                    </Grid>
                </div>
            </div >
        );
    }
}

const mapStateToProps = state => createStructuredSelector(
    {
        photoGrid: state => state.gallery.photoGrid,
        galleryInfo: state => state.gallery.galleryInfo
    });

const mapDispatchToProps = dispatch => {
    return {
        fetchGalleryInformation: id => dispatch({ type: actionType.FETCH_GRID_REQ, param: id })
    }
}

export default withStyles(styles)(connect(mapStateToProps, mapDispatchToProps)(Gallery));