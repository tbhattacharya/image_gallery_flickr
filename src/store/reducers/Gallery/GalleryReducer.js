import { initialState } from '../../initialState/Gallery/initialState';
import * as actionType from '../../actions/Gallery/actionType';

export const GalleryReducer = (state = initialState, action) => {
    switch (action.type) {
        case actionType.FETCH_GRID_SUCCESS:
            return {
                ...state,
                photoGrid: action.photoGrid
            }
        case actionType.UPDATE_GALLERY_INFO:
            return {
                ...state,
                galleryInfo: action.galleryInfo
            }
        default:
            return state;
    }
}