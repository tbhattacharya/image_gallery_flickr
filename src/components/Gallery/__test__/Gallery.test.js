import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { withStyles } from '@material-ui/core/styles';
import Gallery from '../Gallery';
import configureStore from 'redux-mock-store'
import { initialState } from '../../../store/initialState/Gallery/initialState';
import { gallery, photo } from '../../../utils/test-utils/GalleryData';
const mockStore = configureStore()


describe("Testing Gallery", function () {
  const shallow = createShallow();
  const store = mockStore(initialState)
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Gallery store={store} photoGrid= {photo} galleryInfo={gallery}/>)
  })

  it("Expect Gallery to be drawn without exception", function () {
    expect(wrapper).toBeDefined();
  });

  it("Expect Gallery to match snapshot", function () {
    expect(wrapper).toMatchSnapshot();
  });
});