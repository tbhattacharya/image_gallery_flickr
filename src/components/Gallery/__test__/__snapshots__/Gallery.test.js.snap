// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Gallery Expect Gallery to match snapshot 1`] = `
<Connect(Gallery)
  classes={
    Object {
      "mainConatiner": "Connect-Gallery--mainConatiner-02",
      "overlayItem": "Connect-Gallery--overlayItem-04",
      "photo": "Connect-Gallery--photo-06",
      "root": "Connect-Gallery--root-01",
      "subtitle": "Connect-Gallery--subtitle-03",
      "thumbnail": "Connect-Gallery--thumbnail-05",
    }
  }
  galleryInfo={
    Object {
      "count_comments": 69,
      "count_photos": 48,
      "count_total": 48,
      "count_videos": 0,
      "count_views": 38412,
      "date_create": "1480995213",
      "date_update": "1514652568",
      "description": Object {
        "_content": "For the Top Flickr Photos of 2016, we sorted through billions of uploads to arrive at these 50 favorites. 
    
    To compile this list, we started with an algorithm that calculates a combination of social and engagement metrics. To avoid the results being merely a popularity contest, the selection also involved curation by Flickr staff.
    
    Where photos from the same community members appeared multiple times in the algorithmic ranking, we’ve selected one of their top-scoring images to represent their work.
    
    <a href=\\"http://blog.flickr.net/en/2016/12/06/top-flickr-photos-from-2016/\\" rel=\\"nofollow\\">blog.flickr.net/en/2016/12/06/top-flickr-photos-from-2016/</a>",
      },
      "gallery_id": "72157677476597845",
      "iconfarm": 4,
      "iconserver": "3741",
      "id": "66911286-72157677476597845",
      "owner": "66956608@N06",
      "owner_name": "Flickr ",
      "primary_photo_farm": 2,
      "primary_photo_id": "24173905193",
      "primary_photo_secret": "89196bf532",
      "primary_photo_server": "1695",
      "sort_group": null,
      "title": Object {
        "_content": "Top Flickr Photos of 2016",
      },
      "url": "https://www.flickr.com/photos/flickr/galleries/72157677476597845",
      "username": "Flickr",
    }
  }
  photoGrid={
    Array [
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24173905193",
        "is_primary": 1,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "75857967@N08",
        "secret": "89196bf532",
        "server": "1695",
        "title": "Cuando amanece la pasión",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "23971366330",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "127665714@N08",
        "secret": "197b0b3cc5",
        "server": "1675",
        "title": "The Lonely Tree",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24738296325",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "78621811@N06",
        "secret": "9891e3ea56",
        "server": "1618",
        "title": "Zig Zag",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "25432433465",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "47919595@N02",
        "secret": "472cd8795a",
        "server": "1535",
        "title": "Lake Mungo / Mungo National Park",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "28028506096",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "42957889@N05",
        "secret": "f6a6eb1a10",
        "server": "7433",
        "title": "A Wonderful Day",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "27853439812",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "113231763@N07",
        "secret": "1b886779f3",
        "server": "7299",
        "title": "Tropic Thunder II",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24679948091",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "62118446@N03",
        "secret": "dd3e3f117f",
        "server": "1671",
        "title": "Braving the cold",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "26305213826",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "138050950@N05",
        "secret": "5fa1957826",
        "server": "1645",
        "title": "Sound of silence [explored]",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "27190173775",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "130724719@N05",
        "secret": "e9b9533ff4",
        "server": "7156",
        "title": "French Senate from the other side !",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "26694616403",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "137893408@N06",
        "secret": "6904ee640f",
        "server": "7628",
        "title": "On the road",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "28302776541",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "41993130@N04",
        "secret": "6f9c2a6d28",
        "server": "7654",
        "title": "Sunset - Grand Teton National Park",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "25933072772",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "116424075@N06",
        "secret": "bbb1a3e60c",
        "server": "1623",
        "title": "spotlight",
      },
      Object {
        "farm": 9,
        "has_comment": 0,
        "id": "28099637150",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "114727552@N04",
        "secret": "f6872239ba",
        "server": "8397",
        "title": "Changing of the guard",
      },
      Object {
        "farm": 9,
        "has_comment": 0,
        "id": "28040789350",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "55910595@N05",
        "secret": "bed6293793",
        "server": "8853",
        "title": "Tempest || Dubai",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24465232589",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "30125007@N03",
        "secret": "695c05f11e",
        "server": "1636",
        "title": "Bringing back Autumn",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24272982102",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "70992083@N07",
        "secret": "612bf3f919",
        "server": "1571",
        "title": "Beyond Perception",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24881801442",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "91805169@N04",
        "secret": "dd850d575a",
        "server": "1651",
        "title": "Misty Forest",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "23623259713",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "130984525@N03",
        "secret": "27480b6928",
        "server": "1713",
        "title": "Infinity",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "27591124962",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "35591378@N03",
        "secret": "946ba99eb6",
        "server": "7124",
        "title": "P061516PS-0016",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24354970986",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "123495891@N02",
        "secret": "042477cac3",
        "server": "1649",
        "title": "Superman & Lex Luthor",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "25659172370",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "43738607@N07",
        "secret": "ee3b364102",
        "server": "1705",
        "title": "i.get.a.kick.out.of.you",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24246227076",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "136485307@N06",
        "secret": "5f8bc64d74",
        "server": "1523",
        "title": "Apollo 8 Liftoff View",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24519079014",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "57803084@N07",
        "secret": "7ebb3d9af7",
        "server": "1532",
        "title": "",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "25532790890",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "29021614@N05",
        "secret": "c261f84f72",
        "server": "1652",
        "title": "Rainbow Ridge",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "25768216394",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "15599795@N06",
        "secret": "5a2e6a03a7",
        "server": "1491",
        "title": "Freedom !",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "25017672329",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "45992839@N04",
        "secret": "e5b9967612",
        "server": "1659",
        "title": "Red Barn | Norway",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "26092060062",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "68778436@N04",
        "secret": "f2819c0c4e",
        "server": "1587",
        "title": "Flickr Top 50, Top US & Cover Photo 2016 (Fetch - Sunset)",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "27131116894",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "112386875@N05",
        "secret": "465d9d7c8a",
        "server": "7431",
        "title": "Mountain Glory (Explore)",
      },
      Object {
        "farm": 9,
        "has_comment": 0,
        "id": "28257256940",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "96106108@N07",
        "secret": "318ec718f9",
        "server": "8156",
        "title": "frames",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "27783727896",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "52725579@N05",
        "secret": "6e10e7837d",
        "server": "7672",
        "title": "Wallaby Sunset (Explored Jun 21, 2016)",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24890975982",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "66926252@N07",
        "secret": "282deddba9",
        "server": "1599",
        "title": "Bow Valley Sunrise",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "24393016550",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "34430197@N04",
        "secret": "127351d623",
        "server": "1572",
        "title": "Sunset From The Cave",
      },
      Object {
        "farm": 9,
        "has_comment": 0,
        "id": "29214702851",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "22461382@N00",
        "secret": "7feb8d9b34",
        "server": "8107",
        "title": "f o g · f i n g e r s | marin county, california",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "26258214422",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "134276106@N02",
        "secret": "e75fed0ca9",
        "server": "1477",
        "title": "Every Forgotten Dream",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "23822677670",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "83802319@N00",
        "secret": "ea2fc01f71",
        "server": "1616",
        "title": "2016 January Fuji",
      },
      Object {
        "farm": 6,
        "has_comment": 0,
        "id": "30447951876",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "78422747@N04",
        "secret": "dec0cae941",
        "server": "5644",
        "title": "Autumn Arch",
      },
      Object {
        "farm": 9,
        "has_comment": 0,
        "id": "29133157664",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "28164079@N05",
        "secret": "856aef9bc3",
        "server": "8385",
        "title": "Norwegian romantic nationalism",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "26723045205",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "18205706@N06",
        "secret": "6fc0847622",
        "server": "1519",
        "title": "Ballistic",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "23761130783",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "33616145@N00",
        "secret": "843132bdc5",
        "server": "1611",
        "title": "..and then the sky went pink",
      },
      Object {
        "farm": 9,
        "has_comment": 0,
        "id": "29194948535",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "91140701@N03",
        "secret": "48ac690a0c",
        "server": "8544",
        "title": "Kenya sunset (On Explore 08/24/2016)",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "26230731605",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "57866871@N03",
        "secret": "4c10757ae2",
        "server": "1472",
        "title": "Brixham twilight",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "27188420796",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "28319834@N05",
        "secret": "1154494306",
        "server": "7187",
        "title": "Morning reflections",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "26957168394",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "40252249@N00",
        "secret": "30e3ac4250",
        "server": "7452",
        "title": "Welcome to Middle Earth - Explored",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "23958447389",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "76482173@N05",
        "secret": "c6ed281ea5",
        "server": "1567",
        "title": "Watagan Lightning",
      },
      Object {
        "farm": 8,
        "has_comment": 0,
        "id": "26917893065",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "32553078@N08",
        "secret": "0696225a3e",
        "server": "7620",
        "title": "Sunset on Place de l'Etoile after a rainy day",
      },
      Object {
        "farm": 9,
        "has_comment": 0,
        "id": "29558462534",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "86832534@N03",
        "secret": "7ac1239618",
        "server": "8554",
        "title": "Ethereal (Explored)",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "25017412802",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "26248101@N08",
        "secret": "d26a2b206a",
        "server": "1559",
        "title": "Iron Maiden",
      },
      Object {
        "farm": 2,
        "has_comment": 0,
        "id": "25788014884",
        "is_primary": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
        "owner": "130608600@N05",
        "secret": "4d19109166",
        "server": "1633",
        "title": "CRS-8 first stage landing",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
